Leanplum/Api/LPRequest.h

```diff
@@ -0,0 +1,32 @@
+//
+//  LPRequest.h
+//  Leanplum
+//
+//  Created by Grace on 6/10/19.
+//  Copyright © 2019 Leanplum. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <LPApiMethods.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LPRequest : NSObject
+
+@property (nonatomic, strong) NSString *requestId;
+@property (nonatomic, strong) NSString *httpMethod;
+@property (nonatomic, assign) LPApiMethod apiMethod;
+@property (nonatomic, strong, nullable) NSDictionary *params;
+@property (atomic) BOOL sent;
+@property (nonatomic, copy, nullable) void (^successResponse) (NSDictionary *);
+@property (nonatomic, copy, nullable) void (^failureResponse) (NSError *);
+
+- (LPRequest *)initWithApiMethod:(LPApiMethod)apiMethod params:(nullable NSDictionary *)params success:(void (^)(NSDictionary *dictionary))success
```

Comment:
Please align this method: 




- (LPRequest *)initWithApiMethod:(LPApiMethod)apiMethod
                                             params:(nullable NSDictionary *)params 
                                            success:(void (^)(NSDictionary *dictionary))success
                                               failure:(void (^)(NSError *error))failure;

Also did you forget to add httpMethod as a params too?

Reply:
lets return id.

Reply:
`httpMethod` is implicit from the `apiMethod`

@gg4race let's return `instancetype` and not `LPRequest` .. it's conventional obj-c and stronger than `id`